compiler/luci/pass/src/FuseMulWithDivPass.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseMulWithDivPass.h"
+#include "CircleOptimizerUtils.h"
+
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/IR/CircleNodes.h>
+
+namespace luci
+{
+
+/**
+ * Pass to fuse mul(one of the input is const scalar) and
+ * div(numerator is const scalar) as div
+ *
+ * BEFORE
+ *             [CircleNode]                         [Scalar_Mul_Const]
+ *                  |                                      |
+ * [CirlceMul, (x=CircleNode, y=Scalar_Mul_Const)] --------
+ *                  |
+ *                  |                             [Scalar_Div_Const]
+ *                  |                                             |
+ *           [CircleDiv, (x=CirlceMul, y=Scalar_Div_Const] --------
+ *                 |
+ *             [CircleNode]
+ *
+ *  AFTER
+ *          [CircleNode]
+ *                 |                                        [Scalar_new_Div_Const]
+ *                 |                                         |
+ *          [Div, (x=CircleNode, y=Scalar_new_Div_Const] -------
+ *
```

Comment:
1/ please add
```
 *                  |
 *              [CircleNode]
```
2/ let's match `|` and end point of `---`

Reply:
Done